tests/discord/syncRoles.test.ts

```diff
@@ -0,0 +1,35 @@
+import request from 'supertest';
+import app from '../../src/app';
+import { sandbox } from '../setup';
+import * as discordFunctions from '../../src/utils/discord';
+import { Error } from '../../src/types';
+import env from '../../src/utils/env';
+
+// eslint-disable-next-line mocha/no-exclusive-tests
+describe.only('POST /discord/sync-roles', () => {
+  const token = env.discord.syncKey;
+
+  it('should fail because the token is not provided', async () => {
+    await request(app).post('/discord/sync-roles').send({}).expect(401, { error: Error.NoToken });
+  });
```

Comment:
Pour que les tests se terminent bien, il faut retourner une valeur (même si c'est un `null`) dans les promesses *(si on utilise le mot clé `async`)*.
Quand tu n'as qu'une seule opération asynchrone, ce n'est pas la peine d'utiliser async/await (ici `request(...).[...].expect(...)` renvoie déjà une promesse donc on la retourne juste). Je te laisse corriger ça pour les autres tests :wink:
###### *De manière générale, on ne fait jamais de `return await` non plus, puisque ça conserve le stack de la fonction pendant toute la durée du await et que ça exécute une `microTask` (inutile) de plus.*
```suggestion
  it('should fail because the token is not provided', () =>
    request(app).post('/discord/sync-roles').send({}).expect(401, { error: Error.NoToken }));
```